packages/expo-camera/ios/EXCamera/EXCameraManager.m

```diff
@@ -330,6 +338,35 @@ + (NSDictionary *)pictureSizes
   resolve([[[self class] pictureSizes] allKeys]);
 }
 
+UM_EXPORT_METHOD_AS(getAvailableVideoCodecsAsync,
+                    resolve:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+    AVCaptureSession *session = [AVCaptureSession new];
+    
+    [session beginConfiguration];
+
+    NSError *error = nil;
+    AVCaptureDevice *captureDevice = [EXCameraUtils deviceWithMediaType:AVMediaTypeVideo preferringPosition: AVCaptureDevicePositionFront];
+    AVCaptureDeviceInput *captureDeviceInput = [AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:&error];
+    
+    if ([session canAddInput: captureDeviceInput]) {
+        [session addInput: captureDeviceInput];
+    }
+
+    [session commitConfiguration];
+    
+
+    AVCaptureMovieFileOutput *movieFileOutput = [[AVCaptureMovieFileOutput alloc]  init];
+    
+    if ([session canAddOutput: movieFileOutput]) {
+        [session addOutput: movieFileOutput];
+    }
+    
+    resolve([movieFileOutput availableVideoCodecTypes]);
+}
+
+
```

Comment:
```suggestion
UM_EXPORT_METHOD_AS(getAvailableVideoCodecsAsync,
                    resolve:(UMPromiseResolveBlock)resolve
                    rejecter:(UMPromiseRejectBlock)reject)
{
  AVCaptureSession *session = [AVCaptureSession new];   
  [session beginConfiguration];

  NSError *error = nil;
  AVCaptureDevice *captureDevice = [EXCameraUtils deviceWithMediaType:AVMediaTypeVideo preferringPosition: AVCaptureDevicePositionFront];
  AVCaptureDeviceInput *captureDeviceInput = [AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:&error];   
 
  if ([session canAddInput:captureDeviceInput]) {
     [session addInput:captureDeviceInput];
  }

  [session commitConfiguration];

  AVCaptureMovieFileOutput *movieFileOutput = [AVCaptureMovieFileOutput new];
  if ([session canAddOutput:movieFileOutput]) {
    [session addOutput:movieFileOutput];
  }
  
  resolve([movieFileOutput availableVideoCodecTypes]);
}

```

I'm curious about creating the whole AVCaptureSession here 🤔 
Aren't the lines
```
AVCaptureMovieFileOutput *movieFileOutput = [AVCaptureMovieFileOutput new];
resolve([movieFileOutput availableVideoCodecTypes]);
```
covering the whole functionality? 🤔 
I'm looking at the Apple Docs: https://developer.apple.com/documentation/avfoundation/avvideocodectype and https://developer.apple.com/documentation/avfoundation/avcapturemoviefileoutput/1648381-availablevideocodectypes
and wondering if these are not sufficient here? 
Or maybe I'm missing the point and we actually need to create the whole camera session to get the actual data 🤔